runners/direct-java/src/main/java/org/apache/beam/runners/direct/ViewEvaluatorFactory.java

```diff
@@ -77,7 +77,9 @@ public void processElement(WindowedValue<Iterable<InT>> element) {
       @Override
       public TransformResult finishBundle() {
         writer.add(elements);
-        return StepTransformResult.withoutHold(application).build();
+        return StepTransformResult.withoutHold(application)
+            .withAdditionalOutput(!elements.isEmpty())
```

Comment:
This is the only place where the parameter might be `false`. For builder methods, it is certainly better to be able to unset any boolean you can set, so that's good. But it still seems like it might be overgeneralized.